extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/GeneratedStaticResourcesProcessor.java

```diff
@@ -0,0 +1,80 @@
+import io.quarkus.devui.deployment.GeneratedStaticResourcesBuildItem;
+import io.quarkus.vertx.http.deployment.spi.AdditionalStaticResourceBuildItem;
+import io.quarkus.vertx.http.deployment.spi.GeneratedStaticResourceBuildItem;
+import io.quarkus.vertx.http.runtime.GeneratedStaticResourcesRecorder;
+import io.quarkus.vertx.http.runtime.RouteConstants;
+import io.quarkus.vertx.http.runtime.handlers.DevClasspathStaticHandler;
+
+/**
+ * {@link GeneratedStaticResourcesProcessor} is responsible for dealing {@link GeneratedStaticResourceBuildItem}
+ * creating a {@link DevClasspathStaticHandler} to handle all static resources
+ * generated from extensions through {@link GeneratedStaticResourceBuildItem} build item.
+ */
+public class GeneratedStaticResourcesProcessor {
+
+    private static final int ROUTE_ORDER = RouteConstants.ROUTE_ORDER_BEFORE_DEFAULT + 60;
+
+    @BuildStep
+    public void produceResources(List<GeneratedStaticResourceBuildItem> generatedStaticResources,
+            BuildProducer<GeneratedResourceBuildItem> generatedResourceBuildItem,
+            BuildProducer<NativeImageResourceBuildItem> nativeImageResourcesProducer,
+            LaunchModeBuildItem launchModeBuildItem,
+            BuildProducer<AdditionalStaticResourceBuildItem> additionalStaticResourcesProducer,
+            BuildProducer<GeneratedStaticResourcesBuildItem> generatedStaticResourcesProducer) {
+
+        Set<String> generatedResourcesPaths = new HashSet<>();
+        for (GeneratedStaticResourceBuildItem generatedStaticResource : generatedStaticResources) {
+            generatedResourcesPaths.add(generatedStaticResource.getPublicPath());
+            String generatedStaticResourceLocation = buildGeneratedStaticResourceLocation(generatedStaticResource);
+
+            generatedResourceBuildItem.produce(
+                    new GeneratedResourceBuildItem(generatedStaticResourceLocation,
+                            generatedStaticResource.getContent(), false));
+            if (!launchModeBuildItem.getLaunchMode().isDevOrTest()) {
+                additionalStaticResourcesProducer.produce(
+                        new AdditionalStaticResourceBuildItem(generatedStaticResource.getPublicPath(), false));
+                nativeImageResourcesProducer.produce(new NativeImageResourceBuildItem(generatedStaticResource.getPublicPath()));
+            }
+        }
+
+        if (!generatedResourcesPaths.isEmpty()) {
+            generatedStaticResourcesProducer.produce(new GeneratedStaticResourcesBuildItem(generatedResourcesPaths));
+        }
+
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void process(Optional<GeneratedStaticResourcesBuildItem> generatedStaticResources,
+            BuildProducer<RouteBuildItem> routes, GeneratedStaticResourcesRecorder generatedStaticResourcesRecorder,
+            LaunchModeBuildItem launchModeBuildItem) {
+
+        if (generatedStaticResources.isEmpty())
```

Comment:
We usually recommend to have braces even in this case.
It's easy to get things broken if not.